### PR TITLE
Add Filter to Compress HTML

### PR DESCRIPTION
**Pull request by phase**
HTML on all pages will be compressed. This uses the most up to date HTML compression library I found on Maven.

### CODE REVIEW

**Comment by phase** [comment]
[Here](https://gist.github.com/phase/6176b8f399cb1650576e9f1ff95ad57d) is a gist that has a compressed version and an uncompressed version. Going from 18,830 bytes to 9,302 bytes.

/cc @kashike

File: app/util/HtmlCompressionFilter.scala [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+      } else {
+        Future.successful(result)
+      }
+    })
+  }
+
+  private def isCompressible(result: Result): Boolean = {
+    val isChunked = result.header.headers.get(HeaderNames.TRANSFER_ENCODING).contains(HttpProtocol.CHUNKED)
+    val isGzipped = result.header.headers.get(HeaderNames.CONTENT_ENCODING).contains("gzip")
+    !(isChunked || isGzipped)
```
Comment (stephan-gh) [inline app/util/HtmlCompressionFilter.scala:50]: This looks like you avoid compressing the HTML if the response is GZIP compressed. Considering like 100% of all browsers support GZIP, it sounds like the actual issue here is that Ore doesn't compress responses using GZIP.

Comment (Katrix) [reply app/util/HtmlCompressionFilter.scala:50]: Would maybe a better solution be to just add the Gzip filter?
https://www.playframework.com/documentation/2.6.x/GzipEncoding

**Comment by Katrix** [comment]
Will be done on ngnix level instead, or if not that, using the provided play Gzip filter

**Pull request closed without merging**